atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/runner/DefaultWorkflowRunnerTest.java

```diff
@@ -33,25 +35,38 @@
 
 public class DefaultWorkflowRunnerTest {
 
-    private static final ListeningExecutorService EXECUTOR_SERVICE =
-            MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(8));
-
     @Test
     public void runExecutesAllWorkflows() {
-        List<Workflow> workflows = List.of(mock(Workflow.class), mock(Workflow.class));
-        DefaultWorkflowRunner runner = new DefaultWorkflowRunner(EXECUTOR_SERVICE);
-        runner.run(workflows).values().forEach(Futures::getUnchecked);
-        workflows.forEach(workflow -> verify(workflow).run());
+        ListeningExecutorService executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(8));
+        try {
+            List<Workflow> workflows = List.of(mock(Workflow.class), mock(Workflow.class));
+            DefaultWorkflowRunner runner = new DefaultWorkflowRunner(executor);
+            runner.run(workflows).values().forEach(Futures::getUnchecked);
+            executor.shutdown();
+            workflows.forEach(workflow -> verify(workflow).run());
+        } finally {
+            executor.shutdownNow();
+        }
     }
 
     @Test
     public void runMapsHistoryToTheCorrectWorkflow() {
-        Map<Workflow, WorkflowHistory> workflows = Map.of(
-                mock(Workflow.class), mock(WorkflowHistory.class), mock(Workflow.class), mock(WorkflowHistory.class));
-        workflows.entrySet().forEach(entry -> when(entry.getKey().run()).thenReturn(entry.getValue()));
-        DefaultWorkflowRunner runner = new DefaultWorkflowRunner(EXECUTOR_SERVICE);
-        Map<Workflow, ListenableFuture<WorkflowHistory>> histories = runner.run(List.copyOf(workflows.keySet()));
-        assertThat(EntryStream.of(histories).mapValues(Futures::getUnchecked).toMap())
-                .containsExactlyEntriesOf(workflows);
```

Comment:
this was enforcing map entry order, but we're not sorting so its a coin flip if this passes